src/prefect/_internal/schemas/validators.py

```diff
@@ -496,6 +497,35 @@ def validate_trigger_within(
     return value
 
 
+def validate_proactive_trigger_within(values: Dict[str, Any]) -> Dict[str, Any]:
```

Comment:
Can we keep these over in the `events` modules?  I don't like that they're so far away from the schemas that use them.

Reply:
Also, why do we need to move these?  I'm not quite tracking

Reply:
the plan was to consolidate validation where we can; but you may be right that moving the root validators specifically may not be as useful as we think because I don't know that we can use these when creating custom types anyway since they're based on more than a single value. 

cc @zzstoatzz 

I'll close this PR for now.